test/org/sagebionetworks/bridge/validators/AppConfigValidatorTest.java

```diff
@@ -53,8 +68,45 @@ public void before() {
         appConfig = AppConfig.create();
         appConfig.setStudyId(TEST_STUDY_IDENTIFIER);
         
-        this.newValidator = new AppConfigValidator(surveyService, schemaService, Sets.newHashSet("foo","bar"), true);
-        this.updateValidator = new AppConfigValidator(surveyService, schemaService, Sets.newHashSet("foo","bar"), false);
+        this.newValidator = new AppConfigValidator(surveyService, schemaService, appConfigElementService, Sets.newHashSet("foo","bar"), true);
+        this.updateValidator = new AppConfigValidator(surveyService, schemaService, appConfigElementService, Sets.newHashSet("foo","bar"), false);
+    }
+    
+    @Test
+    public void configReferenceValidated() {
+        ConfigReference ref1 = new ConfigReference("id:1", 1L);
+        ConfigReference ref2 = new ConfigReference("id:2", 2L);
+        
+        when(appConfigElementService.getElementRevision(any(), any(), anyLong())).thenReturn(AppConfigElement.create());
+        
+        List<ConfigReference> references = ImmutableList.of(ref1, ref2);
+        appConfig.setConfigReferences(references);
+        
+        // This succeeds because the mock does not throw an exception
+        assertValidatorMessage(newValidator, appConfig, "criteria", "are required");
```

Comment:
This test semantically suggests that the validator should throw because criteria are missing. It also means if there's a bug and the validator throws anyway, we wouldn't catch that bug. It would be better to just call Validate.validateThrowException().

Reply:
Yes, it's confusing. Changed this.